library/ui/src/main/java/com/google/android/exoplayer2/ui/TrackSelectionDialogBuilder.java

```diff
@@ -195,6 +198,12 @@ public TrackSelectionDialogBuilder setShowDisableOption(boolean showDisableOptio
     return this;
   }
 
+  public void setComparator(Comparator<Format> comparator) {
+    if(this.comparator != comparator) {
```

Comment:
There's no real benefit to doing this check, so you can get rid of it and just do `this.comparator = comparator` directly).